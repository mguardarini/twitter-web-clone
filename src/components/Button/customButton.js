import React from 'react';
import Button from '@material-ui/core/Button';
import {withStyles} from '@material-ui/core/styles';

export default function CustomButton(props){
    
const BtnCustom = withStyles(theme => ({
    root: {
      backgroundColor:'#1da1f2',
      '&:hover': {
        backgroundColor: '#1E5DFF',
      }
    },
  }))(Button);

  return (<BtnCustom variant={props.variant} className={props.className} texts={props.text}>
                {props.text}
          </BtnCustom>
  )

}
