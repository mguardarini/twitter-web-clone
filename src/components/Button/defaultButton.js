import React  from 'react';
import Button from '@material-ui/core/Button';

export default function DefaultButton(props)  {
        return (
            <Button 
                color={props.color}
                className={props.className} 
                variant={props.variant} 
                style={props.style} 
                onClick={props.onClick}
            >               
                {props.text}
            </Button>
        )
}


