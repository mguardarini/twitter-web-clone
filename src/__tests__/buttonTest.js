import React from 'react'
import {render, fireEvent} from '@testing-library/react'
import Button from '../../src/components/Button/defaultButton'
 
describe('<Button /> spec', () => {
    it('Renders the component', () => {
      const container = render(<Button text='Nome' />)
      expect(container.firstChild).toMatchSnapshot()
      expect(container.findByLabelText("Nome"))
     })

     it('Should click in button', () => {
        const onClick = jest.fn();
        const {getByText } = render(<Button text='Nome' onClick={onClick} />);    
        const button = getByText ('Nome');
        fireEvent.click(button);
        expect(onClick).toHaveBeenCalled();
    })
});