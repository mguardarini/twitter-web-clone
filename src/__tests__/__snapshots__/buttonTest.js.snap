// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> spec Renders the component 1`] = `undefined`;
